python/ray/ml/preprocessors/imputer.py

```diff
@@ -79,26 +80,19 @@ def __repr__(self):
 def _get_most_frequent_values(
     dataset: Dataset, *columns: str
 ) -> Dict[str, Union[str, Number]]:
-    # TODO(matt): Optimize this.
-    results = {}
-    for column in columns:
-        # Remove nulls.
-        nonnull_dataset = dataset.map_batches(
-            lambda df: df.dropna(subset=[column]), batch_format="pandas"
-        )
-        # Count values.
-        counts = nonnull_dataset.groupby(column).count()
-        # Find max count.
-        max_aggregate = counts.aggregate(Max("count()"))
-        max_count = max_aggregate["max(count())"]
-        # Find values with max_count.
-        most_frequent_values = counts.map_batches(
-            lambda df: df.drop(df[df["count()"] < max_count].index),
-            batch_format="pandas",
-        )
-        # Take first (sorted) value.
-        most_frequent_value_count = most_frequent_values.take(1)[0]
-        most_frequent_value = most_frequent_value_count[column]
-        results[f"most_frequent({column})"] = most_frequent_value
-
-    return results
+    columns = list(columns)
```

Comment:
Nit: I don't think that this list conversion is necessary?
```suggestion
```

Reply:
It is, unfortunately - otherwise it fails with getting the subset of columns from df

Reply:
Ah I see, thanks!